tests/test_pin_job.py

```diff
@@ -36,3 +36,9 @@ def test_clear_pinboard(self, base_url, selenium):
         page.pinboard.clear_pinboard()
         assert page.pinboard.is_pinboard_open
         assert 0 == page.pinboard.pins
+
+    def test_pin_all_jobs(self, base_url, selenium):
+        """ Open treeherder page, pin all jobs, confirm no more than 500 pins in pinboard"""
+        page = TreeherderPage(base_url, selenium).open()
+        page.pin_all_jobs()
+        assert 500 >= page.pinboard.pins
```

Comment:
We can't guarantee there's going to be > 500 jobs. As mentioned in my previous comment, I think a `ResultSet` region would allow us to do more here. For example, the test could look like this:

```python
page.result_sets[0].pin_all_jobs()
assert len(page.result_sets[0].jobs) == len(page.pinboard.pins)
```